File: core/src/nl/t64/game/rpg/components/quest/QuestState.java
package nl.t64.game.rpg.components.quest;

import lombok.AllArgsConstructor;


@AllArgsConstructor
public enum QuestState {
    UNKNOWN(0),
    KNOWN(1),
    ACCEPTED(2),
    UNCLAIMED(3),
    FINISHED(4);

    private final int index;

    public boolean isAtLeast(QuestState otherState) {
        return index >= otherState.index;
    }

    public boolean isAtMost(QuestState otherState) {
        return index <= otherState.index;
    }

}

File: core/src/nl/t64/game/rpg/components/quest/QuestState.kt
package nl.t64.game.rpg.components.quest


enum class QuestState(val index: Int) {
    UNKNOWN(0),
    KNOWN(1),
    ACCEPTED(2),
    UNCLAIMED(3),
    FINISHED(4);

    fun isEqualOrHigherThan(otherState: QuestState): Boolean = index >= otherState.index
    fun isEqualOrLowerThan(otherState: QuestState): Boolean = index <= otherState.index

}
